### PR TITLE
Added new `fastlane search_plugins` command

### PR DESCRIPTION
**Pull request by KrauseFx**
screenshot 2016-05-25 19 42 32

Searching for fastlane plugins right in fastlane. Also tests for the plugin fetcher

### CODE REVIEW

File: fastlane/lib/fastlane/plugins/plugin_search.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      rows = plugins.collect do |current|
+        [
+          current.name,
+          WordWrap.ww(current.info, 50),
+          current.downloads
+        ]
+      end
+
+      params = {}
+      params[:rows] = rows
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/plugin_search.rb]: I want to have headers too

File: fastlane/docs/Plugins.md [hunk 2]
```diff
@@ -29,6 +29,21 @@ In general, we might not accept actions that
...
 `fastlane` is an open platform and we enable every developer to extend it to fit their needs. That's why we built a plugin system that allows you and your company to provide `fastlane` plugins to other `fastlane` users. You have the full power and responsibility of maintaining your plugin and keeping it up to date. This is useful if you maintain your own library or web service, and want to make sure the `fastlane` plugin is always up to date.
 
+### Find a plugin
+
+Head over the [AvailablePlugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/AvailablePlugins.md) for a list of plugins you can use.
+
+List all available plugins using
+
+```
+fastlane search_plugin
```
Comment (mfurtak) [inline fastlane/docs/Plugins.md]: I think the command sound a little more natural called `fastlane search_plugins`

File: fastlane/lib/fastlane/plugins/plugin_search.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      if search_query
+        UI.message "Looking for fastlane plugins containing '#{search_query}'..."
+      else
+        UI.message "Listing all available fastlane plugins"
+      end
+
+      plugins = Fastlane::PluginFetcher.fetch_gems(search_query: search_query)
+
+      if plugins.empty?
+        UI.user_error!("Couldn't find any available fastlane plugins containing '#{search_query}'")
```
Comment (mfurtak) [inline fastlane/lib/fastlane/plugins/plugin_search.rb]: Do we want to `user_error!` here? That will make this count as a failure...

Maybe instead we just print an `error` message and exit normally?

File: fastlane/spec/plugin_fetcher_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        expect(plugin1.downloads).to eq(113)
+        expect(plugin1.info).to eq("experimental fastlane plugin")
+      end
+
+      it "returns a filtered set of plugins when a search query is passed" do
+        plugins = Fastlane::PluginFetcher.fetch_gems(search_query: "yolo")
+        expect(plugins.count).to eq(1)
+        expect(plugins.last.name).to eq("yolokit")
+      end
+    end
```
Comment (mfurtak) [inline fastlane/spec/plugin_fetcher_spec.rb]: Nice!

File: fastlane/fastlane.gemspec [hunk 5]
```diff
@@ -31,6 +31,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'addressable', '~> 2.3' # Support for URI templates
   spec.add_dependency 'multipart-post', '~> 2.0.0' # Needed for uploading builds to appetize
   spec.add_dependency 'xcode-install', '~> 1.4.0' # Needed for xcversion and xcode_install actions
+  spec.add_dependency 'word_wrap', '~> 1.0.0'
```
Comment (mfurtak) [inline fastlane/fastlane.gemspec]: Yay! Some other tables could use some ❤️  from this!

File: fastlane/lib/fastlane/plugins/plugin_search.rb (hunk 3 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/plugin_search.rb]: I think both approaches are valid, nobody is gonna run that command on a CI, so the exit status is not as important. My thought was: We tried to find a certain plugin, but couldn't find it.

File: fastlane/docs/Plugins.md (hunk 2 above)
Comment (KrauseFx) [inline fastlane/docs/Plugins.md]: Okay, changing that 👍

File: fastlane/fastlane.gemspec (hunk 5 above)
Comment (KrauseFx) [inline fastlane/fastlane.gemspec]: Agreed 👍 It's a tiny gem, so the dependency doesn't hurt

**Comment by KrauseFx** [comment]
Just pushed a commit addressing the command name.

**Comment by mfurtak** [comment]
🚢  it!

**Comment by KrauseFx** [comment]
Oh yeah 🚀

**Pull request merged**